diff --git a/src/liquidity_pool.rs b/src/liquidity_pool.rs
index e6e464f..2d88689 100644
--- a/src/liquidity_pool.rs
+++ b/src/liquidity_pool.rs
@@ -27,5 +27,14 @@ impl<T: Config> LiquidityPool<T> {
         self.total_liquidity = self.total_liquidity + liquidity_minted;
     }
 
-    // TODO Function to burn liquidity tokens and update reserves
+    // Function to burn liquidity tokens and update reserves
+    pub fn burn(
+        &mut self,
+        liquidity_burned: AssetBalanceOf<T>,
+        amounts_out: (AssetBalanceOf<T>, AssetBalanceOf<T>),
+    ) {
+        self.reserves.0 = self.reserves.0 - amounts_out.0;
+        self.reserves.1 = self.reserves.1 - amounts_out.1;
+        self.total_liquidity = self.total_liquidity - liquidity_burned;
+    }
 }
